frontend/src/components/ui/BackButton.tsx

```diff
@@ -1,15 +1,23 @@
 "use client";
-import { useSearchParams } from "next/navigation";
+import { useRouter } from "next/navigation";
 import { Button } from "@mui/material";
 
 export default function BackButton() {
-  const searchParams = useSearchParams();
-  const queryParams = searchParams.toString();
-  const backUrl = queryParams ? `/?${queryParams}` : "/";
-  console.log("queryParams: %s", queryParams);
+  const router = useRouter();
+  const referrer = typeof document !== 'undefined' ? document.referrer : ''; // Get the referrer (previous page)
```

Comment:
does the fallback make more sense as null or undefined vs empty string?

Reply:
null is better!